scripts/build/toolchain/termux_step_setup_toolchain_23c.sh

```diff
@@ -0,0 +1,219 @@
+		fi
+		LDFLAGS+=" -Wl,-rpath=$TERMUX_PREFIX/lib"
+	else
+		export CC_FOR_BUILD=$CC
+		# Some build scripts use environment variable 'PKG_CONFIG', so
+		# using this for on-device builds too.
+		export PKG_CONFIG=pkg-config
+	fi
+	export PKG_CONFIG_LIBDIR="$TERMUX_PKG_CONFIG_LIBDIR"
+
+	if [ "$TERMUX_ARCH" = "arm" ]; then
+		# https://developer.android.com/ndk/guides/standalone_toolchain.html#abi_compatibility:
+		# "We recommend using the -mthumb compiler flag to force the generation of 16-bit Thumb-2 instructions".
+		# With r13 of the ndk ruby 2.4.0 segfaults when built on arm with clang without -mthumb.
+		CFLAGS+=" -march=armv7-a -mfpu=neon -mfloat-abi=softfp -mthumb"
+		LDFLAGS+=" -march=armv7-a"
+		export GOARCH=arm
+		export GOARM=7
+	elif [ "$TERMUX_ARCH" = "i686" ]; then
+		# From $NDK/docs/CPU-ARCH-ABIS.html:
+		CFLAGS+=" -march=i686 -msse3 -mstackrealign -mfpmath=sse"
+		# i686 seem to explicitly require -fPIC, see
+		# https://github.com/termux/termux-packages/issues/7215#issuecomment-906154438
+		CFLAGS+=" -fPIC"
+		export GOARCH=386
+		export GO386=sse2
+	elif [ "$TERMUX_ARCH" = "aarch64" ]; then
+		export GOARCH=arm64
+	elif [ "$TERMUX_ARCH" = "x86_64" ]; then
+		export GOARCH=amd64
+	else
+		termux_error_exit "Invalid arch '$TERMUX_ARCH' - support arches are 'arm', 'i686', 'aarch64', 'x86_64'"
+	fi
+
+	# -static-openmp requires -fopenmp in LDFLAGS to work; hopefully this won't be problematic
+	# even when we don't have -fopenmp in CFLAGS / when we don't want to enable OpenMP
+	# We might also want to consider shipping libomp.so instead; since r21
+	LDFLAGS+=" -fopenmp -static-openmp"
+
+	# Android 7 started to support DT_RUNPATH (but not DT_RPATH).
+	LDFLAGS+=" -Wl,--enable-new-dtags"
+
+	# Avoid linking extra (unneeded) libraries.
+	LDFLAGS+=" -Wl,--as-needed"
+
+	# Basic hardening.
+	CFLAGS+=" -fstack-protector-strong"
+	LDFLAGS+=" -Wl,-z,relro,-z,now"
+
+	if [ "$TERMUX_DEBUG_BUILD" = "true" ]; then
+		CFLAGS+=" -g3 -O1"
+		CPPFLAGS+=" -D_FORTIFY_SOURCE=2 -D__USE_FORTIFY_LEVEL=2"
+	else
+		CFLAGS+=" -Oz"
+	fi
+
+	export CXXFLAGS="$CFLAGS"
+	export CPPFLAGS+=" -I${TERMUX_PREFIX}/include"
+
+	# If libandroid-support is declared as a dependency, link to it explicitly:
+	if [ "$TERMUX_PKG_DEPENDS" != "${TERMUX_PKG_DEPENDS/libandroid-support/}" ]; then
+		LDFLAGS+=" -Wl,--no-as-needed,-landroid-support,--as-needed"
+	fi
+
+	export GOOS=android
+	export CGO_ENABLED=1
+	export GO_LDFLAGS="-extldflags=-pie"
+	export CGO_LDFLAGS="${LDFLAGS/-Wl,-z,relro,-z,now/}"
+	CGO_LDFLAGS="${LDFLAGS/-static-openmp/}"
+	export CGO_CFLAGS="-I$TERMUX_PREFIX/include"
+	export RUSTFLAGS="-C link-arg=-Wl,-rpath=$TERMUX_PREFIX/lib -C link-arg=-Wl,--enable-new-dtags"
+
+	export ac_cv_func_getpwent=no
+	export ac_cv_func_endpwent=yes
+	export ac_cv_func_getpwnam=no
+	export ac_cv_func_getpwuid=no
+	export ac_cv_func_sigsetmask=no
+	export ac_cv_c_bigendian=no
+
+	termux_setup_standalone_toolchain_23c
```

Comment:
You can just remove this function and only use the enclosing function. [I spun this off a couple years ago because I needed to setup the standalone toolchain early for the Swift host build](https://github.com/termux/termux-packages/pull/5134/commits/892e1c72c0ea0105feda067222f850db3c1c4dc0), but I no longer use it and no other package in this repo calls this function directly now.

Reply:
Good suggestion, will fix!